### PR TITLE
applications: nrf_desktop: Disable serial driver for DK HW config

### PR DESCRIPTION
**Pull request by pdunaj**
By default DK HW enables serial console which is by default not
used in our project.
For release configurations we also need to make sure that Segger RTT
is disabled.

Jira:DESK-851

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

File: applications/nrf_desktop/configuration/nrf52840_pca10059/app_ZRelease.conf [hunk 1]
```diff
@@ -70,6 +70,11 @@ CONFIG_SYS_PM_POLICY_APP=y
 
 CONFIG_GPIO=y
 
+CONFIG_SERIAL=n
+CONFIG_UART_0_NRF_UARTE=n
+CONFIG_UART_INTERRUPT_DRIVEN=n
+CONFIG_CONSOLE=n
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840_pca10059/app_ZRelease.conf]: Do we want to disable `USE_SEGGER_RTT` here?

File: applications/nrf_desktop/configuration/nrf52840_pca10056/app_ZDebug_keyboard.conf [hunk 2]
```diff
@@ -74,6 +74,12 @@ CONFIG_MPU_STACK_GUARD=y
 
 CONFIG_GPIO=y
 
+CONFIG_SERIAL=n
+CONFIG_UART_0_NRF_UART=n
+CONFIG_CONSOLE=n
+CONFIG_UART_CONSOLE=n
+CONFIG_USE_SEGGER_RTT=n
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840_pca10056/app_ZDebug_keyboard.conf]: I guess we need RTT here

File: applications/nrf_desktop/configuration/nrf52840_pca10059/app_ZReleaseMCUBoot.conf [hunk 3]
```diff
@@ -71,6 +71,11 @@ CONFIG_SYS_PM_POLICY_APP=y
 
 CONFIG_GPIO=y
 
+CONFIG_SERIAL=n
+CONFIG_UART_0_NRF_UARTE=n
+CONFIG_UART_INTERRUPT_DRIVEN=n
+CONFIG_CONSOLE=n
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840_pca10059/app_ZReleaseMCUBoot.conf]: Disable `USE_SEGGER_RTT` ?

File: applications/nrf_desktop/configuration/nrf52840_pca10059/app_ZRelease.conf (hunk 1 above)
Comment (pdunaj) [reply applications/nrf_desktop/configuration/nrf52840_pca10059/app_ZRelease.conf]: It is not enabled by board config so we don't have to

File: applications/nrf_desktop/configuration/nrf52840_pca10059/app_ZReleaseMCUBoot.conf (hunk 3 above)
Comment (pdunaj) [reply applications/nrf_desktop/configuration/nrf52840_pca10059/app_ZReleaseMCUBoot.conf]: board config does not enable it so we don't have to

**Comment by pdunaj** [comment]
Hi @wiminordic , can you give a green tick in this one?

**Pull request merged**